src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -210,15 +233,172 @@ public class DeploymentOptions
                 ObjectType.DatabaseOptions,
                 ObjectType.EventNotifications,
                 ObjectType.ServerRoleMembership,
-                ObjectType.AssemblyFiles,
-        };
+                ObjectType.AssemblyFiles
+            }
+        );
+
+        public DeploymentOptionProperty<bool> AllowExternalLibraryPaths { get; set; }
+
+        public DeploymentOptionProperty<bool> AllowExternalLanguagePaths { get; set; }
+
+        public DeploymentOptionProperty<bool> DoNotEvaluateSqlCmdVariables { get; set; }
+
+        public DeploymentOptionProperty<bool> DisableParallelismForEnablingIndexes { get; set; }
+
+        public DeploymentOptionProperty<bool> DoNotDropWorkloadClassifiers { get; set; }
+
+        public DeploymentOptionProperty<bool> DisableIndexesForDataPhase { get; set; }
+
+        public DeploymentOptionProperty<bool> DoNotDropDatabaseWorkloadGroups { get; set; }
+
+        public DeploymentOptionProperty<bool> HashObjectNamesInLogs { get; set; }
+
+        public DeploymentOptionProperty<bool> IgnoreWorkloadClassifiers { get; set; }
+
+        public DeploymentOptionProperty<bool> IgnoreDatabaseWorkloadGroups { get; set; }
+
+        public DeploymentOptionProperty<bool> IsAlwaysEncryptedParameterizationEnabled { get; set; }
+
+        public DeploymentOptionProperty<bool> PreserveIdentityLastValues { get; set; }
+
+        public DeploymentOptionProperty<bool> RestoreSequenceCurrentValue { get; set; }
+
+        public DeploymentOptionProperty<bool> RebuildIndexesOfflineForDataPhase { get; set; }
+
+        private Dictionary<string, string> _displayNameMapDict;
 
         #endregion
 
+        /// <summary>
+        /// Mapping the DisplayName to the dac deploy option
+        /// Adding new properties here would give easy handling of new option to all extensions
+        /// </summary>
+        private void SetDisplayNameForOption()
+        {
+            #region Display Name and Dac Options Mapping
+            DacDeployOptions d = new DacDeployOptions();
```

Comment:
Seems kind of weird to have this in its own function that creates a temporary object soley to get some name values. You should be able to just create the dictionary where you define it above and use `nameof(DacDeployOptions.<PropName>)` for the keys